app/routes.py

```diff
@@ -202,11 +208,25 @@ def get_main_image():
         return redirect(url_for("user_homepage"))
     return render_template("get-main-image.html", form=form)
 
-
-@app.route("/user_homepage")
+"""
+@app.route("/user_homepage", methods=["GET", "POST"])
+def user_homepage():
+    logout_user()
+    return redirect(url_for("home"))
+"""
+@app.route("/user_homepage", methods=["GET", "POST"])
 @login_required
 def user_homepage():
     print(current_user.name)
+    
+    if request.method == 'POST':
+        data = request.get_json()
+        if data:
+            latitude = data.get('latitude')
+            longitude = data.get('longitude')
+            if latitude and longitude:
+                print(f"{latitude}, {longitude}")

```

Comment:
Imho it's not the reponsibilty of `user_homepade`. I would create a separate route for setting geo data that allow only for post requests e.g. update_geo_data